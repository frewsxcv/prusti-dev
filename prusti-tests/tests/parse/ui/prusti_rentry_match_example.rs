use prusti_contracts::*;

// ignore-test: re-entry not yet implemented
#[ensures(
    match result {
        Ok(x) => (forall(|i: usize| i < s.len() ==> is_digit(s[i]))) && int_to_string(x) == s,
        Err(_) => (forall(|i: usize| i < s.len() ==> !is_digit(s[i])))
    }
)]
fn try_string_to_int(s: &str) -> Result<u32, <u32 as std::str::FromStr>::Err> {
    s.parse::<u32>()
}