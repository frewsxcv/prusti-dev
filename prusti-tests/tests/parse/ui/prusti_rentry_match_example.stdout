Need to bless some output